packages/back-end/src/models/ApiKeyModel.ts

```diff
@@ -284,7 +286,14 @@ export async function getApiKeyByIdOrKey(
   const doc = await ApiKeyModel.findOne(
     id ? { organization, id } : { organization, key }
   );
-  return doc ? toInterface(doc) : null;
+
+  if (!doc) return null;
+
+  const apiKey = toInterface(doc);
+
+  return hasReadAccess(readAccessFilter, [apiKey.project || ""])
```

Comment:
Is an empty array treated the same as `[""]` in the read access logic?  I see a special case at the top of `hasReadAccess` to handle an empty array, but not an array of empty strings.  We should be consistent about how we specify "no projects"

Reply:
I've updated this to read:

```ts
 return keys.filter((k) =>
    hasReadAccess(readAccessFilter, k.project ? [k.project] : [])
  );
  ```
  
Technically, a `[""]` isn't treated the same as a `[]` - a `[""]` would pass the first `if` block.

I can add a special clause to the first block, something like: 
```ts
resourceProjects.every((project) => project === "")
``` 

We can ensure an early exit if a `[""]` is passed in.

Currently though, if a `[""]` is passed in, it'll just act as if it's a project not in the `readAccessFilter.projects` array.

For now, I've updated the `ApiKeyModel` and added the additional clause in the first block.

Reply:
I might recommend a slightly different approach.  Change the method signature of `hasReadAccess` to accept a wider range of values for project:

```ts
function hasReadAccess(filter: ReadAccessFilter, projects: string | string[] | null | undefined);
```

Because this is a utility function that will be called from tons of places, it will be much simpler if you can always just pass in `model.project` or `model.projects` without thinking about handling undefined, empty strings, converting to arrays, etc.

Reply:
10-4. I just updated the `hasReadAccess` method to the proposed signature and updated the function as needed. Please let me know if this is more in-line with what you were thinking.

The changes are in commit https://github.com/growthbook/growthbook/pull/1998/commits/ca832017c1f52b8c738fa2bbfd05783dfd2a61b4